### PR TITLE
Dropdowna11y

### PR DESCRIPTION
**Pull request by pauljadam**
still have to figure to how to make them work when all shown together

### CODE REVIEW

**Comment by davodey** [comment]
I notice that the popup opens on hover, is that intentional?  Also, do we need the dark purple outline around the selected item?  If so can we reduce the stroke size?

http://45.55.127.237/pearson-glp-pl/?p=ui-kit-dropdown

**Comment by davodey** [comment]
Overall great work.  We are almost there.  We just need to solve that issue you pointed out.  

Lets continue to work on this branch and once the above is fixed, well merge it into master.

**Comment by pauljadam** [comment]
@davodey I'm trying to copy this pattern here, http://w3c.github.io/aria-practices/examples/menu-button/menu-button-actions-active-descendant.html,  and they do open it on mouse hover and close on hovering out. I can make it only open on click and enter spacebar key press. I'm not really sure what all the requirements are so I just went with copying the ARIA pattern. 

I think we do need an active element focus style and yes I can change it. I started with Orange :)

**Comment by davodey** [comment]
@pauljadam perfect.  lets open this with a click.

**Comment by pauljadam** [comment]
@davodey ok what about mouse out from the menu button and menu itself? Should it close on mousing out or do they need to click the button to close it again or if they click the body it will close it out?

**Comment by davodey** [comment]
i say mouseoit, when body is clicked would be ideal.

**Comment by pauljadam** [comment]
@davodey improved behavior and design now, still have to get them all working together though

**Comment by davodey** [comment]
Awesome!  I'll pull it down and take a look. I think I know whats happening.

File: pattern-lab/source/js/menu.js [hunk 1]
```diff
@@ -1,36 +1,180 @@
...
+                    menubutton.setAttribute('aria-expanded', 'false'); 
+                    menu.classList.add('hidden');
+                    }
+                });
+    });
+
+        menubuttons.forEach(menubutton => {
+            menubutton.addEventListener('click', event => {
+                const target = event.currentTarget,
+                    menu = target.nextElementSibling.querySelector('[role=menu]'),
```
Comment (davodey) [inline pattern-lab/source/js/menu.js]: @pauljadam  in order to target the correct dropdown we had to change what element was getting selected.  Previously we used 'document.querySelector'  The issue with this is that it looks at the entire document, then finds and returns the FIRST node with the ('[role=menu]');  Which is why the first menu would always open, regardless of what dropdown button was clicked.

We changed the 'document' to 'target.nextElementSibling'.  Now when a button is clicked it looks for the event.target and then queries the menu found in the closest dropdown menu.

Does this make sense?

We still have issues with the menu itself, so well want to do something similar and target the element thats being clicked and traverse from there, not target document

**Comment by StommePoes** [comment]
We have the same issue with id's as the one about document.querySelector-- unless mustache takes care of keeping id's unique? if two dropdowns are on one page they'd have the same id.

In our React components I've been using UUID but if there's a better way for mustache, that'll work.

File: pattern-lab/source/js/menu.js (hunk 1 above)
Comment (pauljadam) [reply pattern-lab/source/js/menu.js]: Thanks Dave! Yes I'm trying to make them all work now when 4 instances are displayed so I'll have to modify some more code based on your suggestion.

**Comment by pauljadam** [comment]
@StommePoes thanks for the reminder! I'm trying to make all the ID's dynamically generated now e.g. menu1 2 3 4 menubutton1 2 3 4 etc. I had the menu item id's down and now need to figure out the rest :)

**Comment by davodey** [comment]
The way I coded it, it's not dependant on Id's.  It looks for all the items with a specific class ('.dropdown'), and if it finds any we run a loop and add event handlers / and target the item within that looped item.

Definitely not the most performant, but for prototyping purposes it works.

**Comment by pauljadam** [comment]
@davodey to meet the ARIA pattern I have to add in the ID references, aria-labelledby on the menu, aria-controls on the menubutton, and aria-activedescendant on the menu. Can't do all of the full ARIA pattern without giving everything unique ID values. It's a fun challenge so I don't mind :)

**Comment by davodey** [comment]
AHHH I see... oh boy, fun task :)

**Comment by pauljadam** [comment]
@davodey @StommePoes I should have all the fixes working now :)

File: pattern-lab/source/js/menu.js [hunk 2]
```diff
@@ -1,36 +1,240 @@
...
 
-            menu.classList.toggle('hidden');
-            event.stopPropagation();
+            }
         })
-    });
 
-    menu.forEach(item => {
+        menubutton.setAttribute('id', 'menubutton' + menuButtonsCount);
+        menuButtonsCount += 1;
```
Comment (davodey) [inline pattern-lab/source/js/menu.js]: Good idea!

Comment (StommePoes) [reply pattern-lab/source/js/menu.js]: yeah I've had to do similar in the past, so long as whatever spits out all of that component can keep track of the counter, it works.

**Pull request merged**